drivers/rf-transceiver/hmc630x/README.rst

```diff
@@ -0,0 +1,385 @@
+HMC630X no-OS Driver
+=====================
+
+Supported Devices
+-----------------
+
+- :adi:`HMC6300`
+- :adi:`HMC6301`
+
+Overview
+--------
+
+The HMC630LP3 and HMC630LP3E are high dynamic range Vector Modulator
+RFICs designed for RF predistortion and feed-forward cancellation
+circuits, as well as RF beam forming and amplitude/phase correction
+circuits. The I and Q ports of the HMC630LP3(E) can be used to
+continuously vary the phase and amplitude of RF signals by up to 360
+degrees and 40 dB, respectively, supporting a 3 dB modulation bandwidth
+of 180 MHz. Key features include continuous phase control of 360°,
+continuous gain control of 40 dB, and a low output noise floor of -162
+dBm/Hz. These characteristics make the device ideal for applications
+such as wireless infrastructure HPA and MCPA error correction and
+cellular/3G systems. Its high input IP3 of +34 dBm and compact 16-lead
+3x3mm SMT format ensure robust performance in advanced RF applications.
+
+Applications
+------------
+
+- Wireless infrastructure HPA and MCPA error correction
+- Pre-distortion or feed-forward linearization
+- Cellular/3G systems
+- Beamforming or RF cancellation circuits
+
+Operation Modes
+---------------
+
++---------------------+------------------------------+-------------------------------+-------------------------------+
+| Mode Name           | Description                  | Configuration Bits            | Typical Use                   |
```

Comment:
this table is full of AI slop, it's really really bad, we have to do something about this

i know it because i wrote these driver files